packages/components/src/unit-control/utils.ts

```diff
@@ -282,32 +288,31 @@ export function filterUnitsWithSettings(
  * @param  args.availableUnits Collection of unit value strings for filtering available units.
  * @param  args.defaultValues  Collection of default values for defined units. Example: { px: '350', em: '15' }.
  *
- * @return Filtered units based on settings.
+ * @return Filtered units of units, with their default values updated following the `defaultValues` argument's property.
  */
 export const useCustomUnits = ( {
-	units,
-	availableUnits,
+	units = ALL_CSS_UNITS,
+	availableUnits = [],
 	defaultValues,
 }: {
-	units?: WPUnitControlUnitList;
-	availableUnits?: Array< string >;
-	defaultValues: Record< string, Value >;
-} ): WPUnitControlUnitList => {
-	units = units || ALL_CSS_UNITS;
-	const usedUnits = filterUnitsWithSettings(
-		! availableUnits ? [] : availableUnits,
+	units?: WPUnitControlUnit[];
+	availableUnits?: string[];
+	defaultValues: Record< string, number >;
```

Comment:
I understand your suggestion, but I think it's a bit tricky since this type is directly tied to the fact that we expect a `quantity` to be of type `number`.

I've taken an alternative approach and:
- refactored all of the `useCustomUnits` instances where `string` where still used instead of `number` (which I had missed until your comment, thank you!)
- I've added a unit test that checks for string values, for added resiliency

What do you think?

Reply:
The looks great to me! Thanks for adding in the unit test and updating those other usages — this gives us the desired behaviour of a consistent API ("this is how the component should be used") while still preserving accepting string values, so it's a nice balance 👍